test/e2e/token_inactivity_timeout_test.go

```diff
@@ -0,0 +1,336 @@
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("token with out timeout should work. Received status %q", resp.Status)
+		}
+	}
+
+	// This checks that
+	// 1. Token with timeout works immediately after they are created
+	// 2. Token with timeout works anytime before it times out
+	// 3. Token with timeout does not work after it times out
+	// 4. Token without timeout works at anytime
+	tokenCheck2 := func(t *testing.T, transport *http.Transport, req *http.Request, tokenWithTimeout, tokenWithoutTimeout *oauthapi.OAuthAccessToken, timeout time.Duration) {
+		req.Header.Set("Authorization", "Bearer "+tokenWithTimeout.Name)
+
+		resp, err := transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("accessing token before it timed out should work. Received status %q", resp.Status)
+		}
+
+		req.Header.Set("Authorization", "Bearer "+tokenWithoutTimeout.Name)
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("token with out timeout should work. Received status %q", resp.Status)
+		}
+
+		time.Sleep(120 * time.Second)
+		req.Header.Set("Authorization", "Bearer "+tokenWithTimeout.Name)
+
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("accessing token before it timed out should work. Received status %q", resp.Status)
+		}
+
+		req.Header.Set("Authorization", "Bearer "+tokenWithoutTimeout.Name)
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("token with out timeout should work. Received status %q", resp.Status)
+		}
+
+		time.Sleep(timeout + 10*time.Second)
+		req.Header.Set("Authorization", "Bearer "+tokenWithTimeout.Name)
+
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusUnauthorized {
+			t.Fatalf("accessing token after it timed out should not work. Received status %q", resp.Status)
+		}
+
+		req.Header.Set("Authorization", "Bearer "+tokenWithoutTimeout.Name)
+		resp, err = transport.RoundTrip(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("token with out timeout should work. Received status %q", resp.Status)
+		}
+	}
+
+	// No OAuthClient timeout and no OAuth config timeout.
+	checkTokenAccess(t, kubeConfig, userClient, oauthClientClient, 10, 0, "without-inactivity-timeout", tokenCheck1)
+
+	// Update OAuth cluster config with token inactivity timeout.
+	updateOAuthConfigInactivityTimeout(t, configClient, &metav1.Duration{Duration: 600 * time.Second})
+	// Wait for kube-apiserver pods to start progressing.
```

Comment:
these comments disturb reading flow more than they help. They don't add anything that's not clear from the func name.